### PR TITLE
✨ Create APIs for MergedPRs and DefaultBranch

### PR DESCRIPTION
**Pull request by azeemshaikh38**
* **Please check if the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] PR title follows the guidelines defined in  https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Adds new APIs for PRs and DefaultBranch

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No.

### CODE REVIEW

File: clients/githubrepo/graphql.go [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+	vars := map[string]interface{}{
+		"owner":                 githubv4.String(owner),
+		"name":                  githubv4.String(repo),
+		"pullRequestsToAnalyze": githubv4.Int(pullRequestsToAnalyze),
+		"reviewsToAnalyze":      githubv4.Int(reviewsToAnalyze),
+		"labelsToAnalyze":       githubv4.Int(labelsToAnalyze),
+	}
+	handler.data = new(graphqlData)
+	if err := handler.client.Query(ctx, handler.data, vars); err != nil {
+		return fmt.Errorf("error querying Github graphQL data: %w", err)
```
Comment (laurentsimon) [inline clients/githubrepo/graphql.go:80]: use the new error handling function `sce.Create()`

File: clients/githubrepo/client.go [hunk 2]
```diff
@@ -55,13 +62,25 @@ func (client *Client) GetFileContent(filename string) ([]byte, error) {
 	return client.tarball.getFileContent(filename)
 }
 
+func (client *Client) ListMergedPRs() []clients.PullRequest {
```
Comment (laurentsimon) [inline clients/githubrepo/client.go:65]: shall we have these function return `error` as well? Same question for `GetDefaultBranch `

File: clients/repo_client.go [hunk 3]
```diff
@@ -41,5 +42,37 @@ type RepoClient interface {
 	InitRepo(owner, repo string) error
 	ListFiles(predicate func(string) bool) []string
 	GetFileContent(filename string) ([]byte, error)
+	ListMergedPRs() []PullRequest
```
Comment (laurentsimon) [inline clients/repo_client.go:45]: see question above about returning an `error`.

**Review by laurentsimon — APPROVED** [review]

File: clients/githubrepo/client.go (hunk 2 above)
Comment (azeemshaikh38) [reply clients/githubrepo/client.go]: Sure, done.

File: clients/githubrepo/graphql.go (hunk 1 above)
Comment (azeemshaikh38) [reply clients/githubrepo/graphql.go]: Done.

File: clients/repo_client.go (hunk 3 above)
Comment (azeemshaikh38) [reply clients/repo_client.go]: Done.

**Pull request merged**